testpar/t_subfiling_vfd.c

```diff
@@ -2414,7 +2487,30 @@ main(int argc, char **argv)
     if (num_iocs_g > mpi_size)
         num_iocs_g = mpi_size;
 
+#ifdef H5_HAVE_FILTER_DEFLATE
+    if (MAINPROCESS) {
+        printf(" Re-running tests with compression enabled\n");
```

Comment:
I would have these prints outside the #ifdef and then just print the SKIPPED if deflate isn't available

Reply:
I tried that, but it would add the "skip" would not line up with the "passed", it would print it after the line. This is what it looks like:

`  Testing file creation and immediate close                            PASSED
  Testing invalid use of IOC VFD by itself                             PASSED
  Testing subfiling configuration file format                          PASSED
  Testing random subfiling stripe sizes                                PASSED
  Testing I/O concentrator selection strategies                        PASSED
  Testing file re-opening with different stripe size                   PASSED
  Testing file pre-creation on rank 0                                  PASSED
  Testing reading back file with single MPI rank                       PASSED
  Testing reading back file with fewer MPI ranks than written with     PASSED
  Testing h5fuse utility                                               PASSED

  Testing re-running tests with compression enabled                    -SKIP-

Re-running tests with environment variables set
  Testing file creation and immediate close                            PASSED
  Testing invalid use of IOC VFD by itself                             PASSED
  Testing subfiling configuration file format                          PASSED
  Testing random subfiling stripe sizes                                PASSED
  Testing I/O concentrator selection strategies                        PASSED
  Testing file re-opening with different stripe size                   PASSED
  Testing file pre-creation on rank 0                                  PASSED
  Testing reading back file with single MPI rank                       PASSED
  Testing reading back file with fewer MPI ranks than written with     PASSED
  Testing h5fuse utility                                               PASSED

  Testing re-running tests with compression enabled                    -SKIP-
`
Except they are all aligned.

Reply:
Typically I would either not print the "re-running" portion if all tests are going to be skipped, or just let the skipped part be below the line, since the output is just a header really and not a "test" being run

Reply:
This is fine though and makes the output nicely aligned, just a bit awkward working around the #ifdefs

Reply:
Changed to print skip after line:
`
Testing Subfiling VFD functionality
Using seed: 1698252665

  Testing file creation and immediate close                            PASSED
  Testing invalid use of IOC VFD by itself                             PASSED
  Testing subfiling configuration file format                          PASSED
  Testing random subfiling stripe sizes                                PASSED
  Testing I/O concentrator selection strategies                        PASSED
  Testing file re-opening with different stripe size                   PASSED
  Testing file pre-creation on rank 0                                  PASSED
  Testing reading back file with single MPI rank                       PASSED
  Testing reading back file with fewer MPI ranks than written with     PASSED
  Testing h5fuse utility                                               PASSED

 Re-running tests with compression enabled
 -SKIP-

Re-running tests with environment variables set
  Testing file creation and immediate close                            PASSED
  Testing invalid use of IOC VFD by itself                             PASSED
  Testing subfiling configuration file format                          PASSED
  Testing random subfiling stripe sizes                                PASSED
  Testing I/O concentrator selection strategies                        PASSED
  Testing file re-opening with different stripe size                   PASSED
  Testing file pre-creation on rank 0                                  PASSED
  Testing reading back file with single MPI rank                       PASSED
  Testing reading back file with fewer MPI ranks than written with     PASSED
  Testing h5fuse utility                                               PASSED

 Re-running tests with compression enabled
 -SKIP-

All Subfiling VFD tests passed

`